src/Controller/ApiV1Controller.php

```diff
@@ -0,0 +1,211 @@
+        $this->disableAutoRender();
+
+        if (!Configure::read('Search.enabled')) {
+            $this->render('search_disabled');
+            return;
+        }
+
+        /* Apply search criteria and sort */
+        $search = new SentencesSearchForm();
+        $search->setData($this->request->getQueryParams());
+
+        /* Control input */
+        if ($search->generateRandomSeedIfNeeded()) {
+            return $this->redirect(Router::url($search->getData()));
+        }
+        $search->checkUnwantedCombinations();
+
+        $limit = CurrentUser::getSetting('sentences_per_page');
+        $sphinx = $search->asSphinx();
+        $sphinx['page'] = $this->request->query('page');
+//        $sphinx['limit'] = $limit;
+
+        // TODO: improve on hard-coded limit to the number of API results
+        $sphinx['limit'] = 100; // Limit the API to returning 100 entries.
+
+        $model = 'Sentences';
+        $query = $this->Sentences
+            ->find('hideFields')
+            ->find('withSphinx')
+            ->find('nativeMarker')
+            ->find('filteredTranslations', [
+                'translationLang' => $search->getData('to'),
+            ])
+            ->select($this->Sentences->fields())
+            ->contain($this->Sentences->contain(['translations' => true]));
+
+        $this->paginate = [
+            'limit' => $limit,
+            'sphinx' => $sphinx,
+        ];
+
+        try {
+            $results = $this->paginate($query);
+            $real_total = $this->Sentences->getRealTotal();
+            $results = $this->Sentences->addHighlightMarkers($results);
+            $this->set(compact('results', 'real_total'));
+        } catch (Exception $e) {
+            $syntax_error = strpos($e->getMessage(), 'syntax error,') !== FALSE;
+            if ($syntax_error) {
+                $this->set('syntax_error', true);
+            } else {
+                $this->loadComponent('Error');
+                $error_code = $this->Error->traceError('Search error: ' . $e->getMessage());
+                $this->set('error_code', $error_code);
+            }
+        }
+
+        $strippedQuery = preg_replace('/"|=/', '', $search->getData('query'));
+        $this->loadModel('Vocabulary');
+        $vocabulary = $this->Vocabulary->findByText($strippedQuery);
+
+        $searchableLists = $search->getSearchableLists(CurrentUser::get('id'));
+        $ignored = $search->getIgnoredFields();
+
+        $this->set($search->getData());
+        $this->set(compact('ignored', 'searchableLists', 'vocabulary'));
+        $this->set(
+            'is_advanced_search',
+            !is_null($this->request->getQuery('trans_to'))
+        );
+
+        $ids = array();
+        foreach ($results as $sentence) {
+            $ids[] = $sentence->id;
+        }
+    
+        $ids_json = json_encode($ids);
+        echo '{ "ids" : ' . $ids_json . ' }';
```

Comment:
Using `echo` here prevents this method from being properly tested with a unit test. You should do it the CakePHP way, for example like you did in the `sentence()` method.

Reply:
Removed echo, changed to:
![image](https://user-images.githubusercontent.com/7943077/117721032-ab9dc900-b1df-11eb-8432-691c6d2fb4e1.png)